ams/server/src/main/java/com/netease/arctic/server/manager/KubernetesOptimizerContainer.java

```diff
@@ -0,0 +1,155 @@
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void requestResource(Resource resource) {
+    // generate pod start args
+    long memoryPerThread =
+        Long.parseLong(checkAndGetProperty(resource.getProperties(), MEMORY_PROPERTY));
+    long memory = memoryPerThread * resource.getThreadCount();
+    // point at amoro home in docker image
+    String startUpArgs =
+        String.format(
+            "/entrypoint.sh optimizer %s %s",
+            memory, super.buildOptimizerStartupArgsString(resource));
+    LOG.info("Starting k8s optimizer using k8s client with start command : {}", startUpArgs);
+
+    String namespace = resource.getProperties().getOrDefault(NAMESPACE, "default");
+    String image = checkAndGetProperty(resource.getProperties(), IMAGE);
+    String cpu = checkAndGetProperty(resource.getProperties(), CPU_PROPERTY);
+
+    String resourceId = resource.getResourceId();
+    String groupName = resource.getGroupName();
+
+    Deployment deployment =
+        new DeploymentBuilder()
+            .withNewMetadata()
+            .withName(NAME + "-" + resourceId)
+            .endMetadata()
+            .withNewSpec()
+            .withReplicas(1)
+            .withNewTemplate()
+            .withNewMetadata()
+            .addToLabels("app", NAME + "-" + resourceId)
+            .addToLabels("AmoroOptimizerGroup", groupName)
+            .addToLabels("AmoroResourceId", resourceId)
+            .endMetadata()
+            .withNewSpec()
+            .addNewContainer()
+            .withName("optimizer")
+            .withImage(image)
+            .withCommand("sh", "-c", startUpArgs)
+            .withResources(
+                new ResourceRequirementsBuilder()
+                    .withLimits(
+                        ImmutableMap.of(
+                            "memory", new Quantity(memory + "Mi"), "cpu", new Quantity(cpu)))
+                    .withRequests(
+                        ImmutableMap.of(
+                            "memory", new Quantity(memory + "Mi"), "cpu", new Quantity(cpu)))
+                    .build())
+            .endContainer()
+            .endSpec()
+            .endTemplate()
+            .withNewSelector()
+            .addToMatchLabels("app", NAME + "-" + resourceId)
+            .endSelector()
+            .endSpec()
+            .build();
+    client.apps().deployments().inNamespace(namespace).resource(deployment).create();
+  }
+
+  @Override
+  protected Map<String, String> doScaleOut(Resource resource) {
+    return null;
+  }
+
+  @Override
+  public void releaseOptimizer(Resource resource) {
+    String resourceId = resource.getResourceId();
+    LOG.info("release Kubernetes Optimizer Container {}", resourceId);
+    client
+        .apps()
+        .deployments()
+        .inNamespace(resource.getGroupName())
+        .withName(NAME + "-" + resourceId)
```

Comment:
I suggest that you store the namespace and deployment's name information in the `doScaleOut`'s return and read it from resource properties during the relesae optimzier, so that there is better compatibility in future code refactoring.